Bottom: 35097adeba80bcc566557dbce6388ba41e097d0d
Top:    aa91824495a842717965ba26da12a43670ba0c21
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-02-08 17:23:13 +0800

Refresh of vhost-packed-ring-support

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index e150648..06f670e 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -1150,13 +1150,13 @@ static int vhost_iotlb_miss(struct vhost_virtqueue *vq, u64 iova, int access)
 }
 
 static int vq_access_ok_packed(struct vhost_virtqueue *vq, unsigned int num,
-			       struct vring_desc_packed __user *desc,
+			       struct vring_desc __user *desc,
 			       struct vring_avail __user *avail,
 			       struct vring_used __user *used)
 {
-	return access_ok(VERIFY_READ | VERIFY_WRITE, desc, num *
-			 sizeof *desc) &&
-		
+	struct vring_desc_packed *packed = (struct vring_desc_packed *)desc;
+	return access_ok(VERIFY_READ, packed, num * sizeof *packed) &&
+	       access_ok(VERIFY_WRITE, packed, num * sizeof *packed);
 }
 
 static int vq_access_ok_split(struct vhost_virtqueue *vq, unsigned int num,
@@ -1179,8 +1179,8 @@ static int vq_access_ok(struct vhost_virtqueue *vq, unsigned int num,
 			struct vring_avail __user *avail,
 			struct vring_used __user *used)
 {
-	if (vhost_has_feature(vq, VIRTIO_RING_F_PACKED))
-		return vq_access_ok_packed();
+	if (vhost_has_feature(vq, VIRTIO_F_RING_PACKED))
+		return vq_access_ok_packed(vq, num, desc, avail, used);
 	else
 		return vq_access_ok_split(vq, num, desc, avail, used);
 }
